ui/js/perf.js

```diff
@@ -170,12 +170,18 @@ perf.factory('PhSeries', ['$http', 'thServiceDomain', function($http, thServiceD
             return _getAllSeries(projectName, timeRange, optionMap).then(function(lists) {
                 lists.seriesList.forEach(function(seriesSummary) {
                     // Only keep summary signatures, filter in/out e10s
-
-                    if (!seriesSummary.subtestSignatures ||
-                        (userOptions.e10s && !_.contains(seriesSummary.options, 'e10s')) ||
+                    if ((userOptions.e10s && !_.contains(seriesSummary.options, 'e10s')) ||
                         (!userOptions.e10s && _.contains(seriesSummary.options, 'e10s'))) {
                         return;
                     } else {
+                        // We should also filter out those subtest series but keep none summary one.
+                        if (!seriesSummary.subtestSignatures) {
+                            if( _.each(lists.seriesList, function(series) {
```

Comment:
_.each isn't going to do you much good here, as it doesn't return anything. You're probably looking for "any" or "some":

https://lodash.com/docs#some

Better still would be to create a list of subtest signatures at the beginning of this function, something like:

    var allSubtestSignatures = _.flatten(_.map(list.seriesList, function(series) { if (series.subtestSignatures) return subtestSignatures; return []));

Then you could just do something like:

    if (_.contains(allSubtestSignatures), series.signatures))
      return;